### PR TITLE
 ec2_vpc_subnet: Add support for OutpostArn param 

### LINKED ISSUE #375: Add support for OutpostArn in ec2_vpc_subnet
##### SUMMARY
Add support for OutpostArn in ec2_vpc_subnet

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
ec2_vpc_subnet

##### ADDITIONAL INFORMATION
Add a module option for OutpostArn in the ec2_vpc_subnet module. When creating a subnet in an Outposts the OutpostARN is needed.

### PR DESCRIPTION
**Pull request by mandar242**
##### SUMMARY

Added support for ```OutpostArn``` param in ```ec2_vpc_subnet```.

Fixes #375 .

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

```ec2_vpc_subnet```

### CODE REVIEW

**Comment by alinabuzachis** [comment]
@mandar242 Thank you for working on this. This issue and https://github.com/ansible-collections/amazon.aws/pull/597 are already on Jira.

File: plugins/modules/ec2_vpc_subnet.py [hunk 1]
```diff
@@ -279,6 +285,9 @@ def create_subnet(conn, module, vpc_id, cidr, ipv6_cidr=None, az=None, start_tim
     if az:
         params['AvailabilityZone'] = az
 
+    if outpost_arn:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_vpc_subnet.py:288]: @mandar242 you can use the same outpost validation function you used for `ec2_vol` module  here.

File: tests/unit/module_utils/test_ec2.py [hunk 2]
```diff
@@ -76,3 +78,19 @@ def test_ansible_dict_with_integer_to_boto3_filter_list(self):
 
         converted_filters_int = ansible_dict_to_boto3_filter_list(filters)
         self.assertEqual(converted_filters_int, filter_list_integer)
+
```
Comment (alinabuzachis) [inline tests/unit/module_utils/test_ec2.py:81]: @mandar242  No need to add the units and the test function inside plugins/module_utils/ec2.py  here. We are going to merge https://github.com/ansible-collections/amazon.aws/pull/597 first.

File: tests/unit/requirements.txt [hunk 3]
```diff
@@ -4,3 +4,4 @@ boto3
 boto
 
 placebo
+parametrize
```
Comment (jillr) [inline tests/unit/requirements.txt:7]: We shouldn't need this any longer if the tests are using @pytest.mark.parametrize now

**Review by jillr — APPROVED** [review]

**Pull request merged**